File: backend/app/schemas/governance.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, ConfigDict

# =======================
# Column Schemas
# =======================
class GovernanceColumnBase(BaseModel):
    column_name: str
    data_type: str
    description: Optional[str] = None
    is_primary_key: bool = False
    is_nullable: bool = True
    security_level: Optional[str] = None
    data_standard: Optional[str] = None

class GovernanceColumnCreate(GovernanceColumnBase):
    pass

class GovernanceColumnUpdate(BaseModel):
    description: Optional[str] = None
    security_level: Optional[str] = None
    data_standard: Optional[str] = None

class GovernanceColumn(GovernanceColumnBase):
    id: int
    table_id: int
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

# =======================
# Table Schemas
# =======================
class GovernanceTableBase(BaseModel):
    schema_name: str
    table_name: str
    description: Optional[str] = None
    owner: Optional[str] = None
    classification_level: Optional[str] = None
    retention_period: Optional[int] = None
    tags: Optional[str] = None

class GovernanceTableCreate(GovernanceTableBase):
    datasource_id: int
    columns: Optional[List[GovernanceColumnCreate]] = None

class GovernanceTableUpdate(BaseModel):
    description: Optional[str] = None
    owner: Optional[str] = None
    classification_level: Optional[str] = None
    retention_period: Optional[int] = None
    tags: Optional[str] = None

class GovernanceTable(GovernanceTableBase):
    id: int
    datasource_id: int
    created_at: datetime
    updated_at: datetime
    columns: List[GovernanceColumn] = []
    
    model_config = ConfigDict(from_attributes=True)

class GovernanceTableList(GovernanceTableBase):
    """列表视图，不包含字段详情"""
    id: int
    datasource_id: int
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

File: dc_frontend/src/types/governance.ts
export interface GovernanceColumn {
  id: number
  table_id: number
  column_name: string
  data_type: string
  is_primary_key: boolean
  is_nullable: boolean
  description?: string
  security_level?: string
  data_standard?: string
  data_quality_rule?: string
  tags?: string[]
  updated_at: string
  updated_by?: number
}

export interface GovernanceTable {
  id: number
  datasource_id: number
  schema_name: string
  table_name: string
  description?: string
  owner?: string
  classification_level?: string
  retention_period?: string
  tags?: string[]
  row_count?: number
  storage_size?: number
  last_analyzed?: string
  created_at: string
  updated_at: string
  created_by?: number
  updated_by?: number
  columns?: GovernanceColumn[]
}

export interface GovernanceTableListQuery {
  page?: number
  size?: number
  search?: string
  datasource_id?: number
}

export interface GovernanceTableUpdate {
  description?: string
  owner?: string
  classification_level?: string
  retention_period?: string
  tags?: string[]
}

export interface GovernanceColumnUpdate {
  description?: string
  security_level?: string
  data_standard?: string
  tags?: string[]
}
